moveit_py/src/moveit/moveit_core/robot_state/robot_state.cpp

```diff
@@ -0,0 +1,507 @@
+           If frame_id was not found, frame_found is set to false and an identity transform is returned.
+       This method is restricted to frames defined within the robot state and doesn't include collision object present in the collision world. Please use the PlanningScene.get_frame_transform method for collision world objects.
+           Args:
+               frame_id (str): The id of the frame to get the transform for.
+           Returns:
+               :py:class:`numpy.ndarray`: The transformation matrix from the model frame to the frame identified by frame_id.
+           )")
+
+      .def("get_pose", &moveit_py::bind_robot_state::get_pose, py::arg("link_name"),
+           R"(
+           Get the pose of a link that is defined in the robot model.
+           Args:
+               link_name (str): The name of the link to get the pose for.
+           Returns:
+               geometry_msgs.msg.Pose: A ROS geometry message containing the pose of the link.
+           )")
+
+      .def("get_jacobian",
+           py::overload_cast<std::shared_ptr<moveit::core::RobotState>&, const std::string&, const Eigen::Vector3d&>(
+               &moveit_py::bind_robot_state::get_jacobian),
+           py::arg("joint_model_group_name"), py::arg("reference_point_position"), py::return_value_policy::move,
+           R"(
+           Compute the Jacobian with reference to the last link of a specified group.
+           Args:
+               joint_model_group_name (str): The name of the joint model group to compute the Jacobian for.
+               reference_point_position (:py:class:`numpy.ndarray`): The position of the reference point in the link frame.
+           Returns:
+               :py:class:`numpy.ndarray`: The Jacobian of the specified group with respect to the reference point.
+           Raises:
+               Exception: If the group is not a chain.
+           )")
+
+      .def("get_jacobian",
+           py::overload_cast<std::shared_ptr<moveit::core::RobotState>&, const std::string&, const std::string&,
+                             const Eigen::Vector3d&, bool>(&moveit_py::bind_robot_state::get_jacobian),
+           py::arg("joint_model_group_name"), py::arg("link_name"), py::arg("reference_point_position"),
+           py::arg("use_quaternion_representation") = false, py::return_value_policy::move,
+           R"(
+           Compute the Jacobian with reference to a particular point on a given link, for a specified group.
+           Args:
+               joint_model_group_name (str): The name of the joint model group to compute the Jacobian for.
+               link_name (str): The name of the link model to compute the Jacobian for.
+               reference_point_position (:py:class:`numpy.ndarray`): The position of the reference point in the link frame.
+           use_quaternion_representation (bool): If true, the Jacobian will be represented as a quaternion.
```

Comment:
indent

And if false, what will it be?

Reply:
I think the default is axis-angle representation. I checked the source code and clearly this flag alters the dimension of the transformation matrix to accommodate a quaternion when it evaluates to true.

Reply:
Oops I am wrong it looks like Euler angles.

Reply:
7a13551